src/mcp/types.py

```diff
@@ -36,7 +36,7 @@
 ProgressToken = str | int
 Cursor = str
 Role = Literal["user", "assistant"]
-RequestId = Annotated[int | str, Field(union_mode="left_to_right")]
+RequestId = Annotated[int, Field(strict=True)] | str
```

Comment:
This just makes sure that only `int` that is an `int` is actually an `int`. 👀

i.e. a string that looks like an int doesn't become an `int`.